src/store/mutations.js

```diff
@@ -295,6 +334,104 @@ function startSilentRequest(state, payload) {
   updateRequest(state, Object.assign(payload, { isSilent: true }), requests.PENDING);
 }
 
+function SOCKET_ONCLOSE(state) {
+  state.socket.client = null;
+  state.socket.isConnected = false;
+  if (!state.socket.connectionClosed) {
+    state.socket.connectionClosed = moment().utc();
+  }
+}
+
+function SOCKET_ONOPEN(state, event) {
+  state.socket.client = event.target;
+  state.socket.isConnected = true;
+  state.socket.connectionClosed = null;
+  if (state.socket.opened) {
+    state.socket.opened();
+  }
+  if (state.needsWsReconnectHandling) {
+    state.needsWsReconnectHandling = false;
+    if (state.currentMarket) {
+      this.dispatch('subscribeToMarket', {
+        market: state.currentMarket,
+        isRequestSilent: true,
+      });
+
+      // Ensure trade history is up-to-date on reconnect. (may have dropped some trades during disconnect)
+      this.dispatch('fetchTradeHistory', {
+        marketName: state.currentMarket.marketName,
+        isRequestSilent: true,
+      });
+    }
+  }
+}
+
+function SOCKET_ONMESSAGE(state, message) {
+  state.socket.lastMessage = message;
+
+  if (message.subscribe && message.subscribe.indexOf('orderBook') > -1) {
+    state.socket.subscribedMarket = message.subscribe.substring(message.subscribe.indexOf(':') + 1);
+  } else if (message.unsubscribe && message.unsubscribe.indexOf('orderBook') > -1) {
+    state.socket.subscribedMarket = null;
+  } else if (message.type === 'bookSnapshot') {
+    orderBookSnapshotReceived(state, message);
+  } else if (message.type === 'bookUpdate') {
+    orderBookUpdateReceived(state, message);
+  } else if (message.type === 'orderCreated') {
+    if (state.socket.orderCreated) {
+      state.socket.orderCreated(message);
+    }
+  } else if (message.type === 'orderMatched') {
+    if (state.socket.orderMatched) {
+      state.socket.orderMatched(message);
+    }
+  } else if (message.type === 'orderCreationFailed') {
+    if (state.socket.orderCreationFailed) {
+      state.socket.orderCreationFailed(message);
+    }
+  } else if (message.type === 'orderMatchFailed') {
+    if (state.socket.orderMatchFailed) {
+      state.socket.orderMatchFailed(message);
+    }
+  } else if (message.type === 'trades') {
+    tradeUpdateReceived(state, message);
+  } else if (message.type) {
+    // unknown message type
+    // console.log(message);
```

Comment:
??

Reply:
I'm fine with keeping the comment, when we add new messages, it might be useful to re-enable the log; though it would probably be more useful if it were `console.log(JSON.stringify(message))`.